### PR TITLE
Fixed nested slices issue, add support to custom datatype prefix

### PR DESCRIPTION
**Pull request by isuruh15**

### CODE REVIEW

File: native/fhir-to-bal-lib/src/main/java/org/wso2/healthcare/fhir/ballerina/packagegen/tool/ToolConstants.java [hunk 1]
```diff
@@ -39,6 +39,10 @@ public class ToolConstants {
     public static final String CONFIG_PACKAGE_REPOSITORY = "repository";
     public static final String CONFIG_PACKAGE_REPOSITORY_TOML = "tools.config.package.repository";
     public static final String CONFIG_BASE_PACKAGE = "basePackage";
```
Comment (sameeragunarathne) [inline native/fhir-to-bal-lib/src/main/java/org/wso2/healthcare/fhir/ballerina/packagegen/tool/ToolConstants.java:41]: Can't we use base package for this?

File: native/fhir-to-bal-lib/src/main/java/org/wso2/healthcare/fhir/ballerina/packagegen/tool/modelgen/ResourceContextGenerator.java [hunk 2]
```diff
@@ -135,7 +136,16 @@ private void populateResourceTemplateContexts(FHIRImplementationGuide ig) {
 
     private void populateElementDefinitionMap(List<ElementDefinition> elementDefinitions) {
         for (ElementDefinition elementDefinition : elementDefinitions) {
-            this.resourceTemplateContextInstance.getSnapshotElementDefinitions().put(elementDefinition.getId(), elementDefinition);
+            String id = elementDefinition.getId();
+            if (id.contains(":")) {
+                if (id.substring(id.indexOf("."), id.lastIndexOf(":")).contains(":")) {
```
Comment (sameeragunarathne) [inline native/fhir-to-bal-lib/src/main/java/org/wso2/healthcare/fhir/ballerina/packagegen/tool/modelgen/ResourceContextGenerator.java:141]: Can't use StringUtils.countMatches(input, ":") ?

File: native/health-cli/src/main/java/io/ballerina/health/cmd/fhir/FhirSubCmd.java [hunk 3]
```diff
@@ -79,6 +79,9 @@ public class FhirSubCmd implements BLauncherCmd {
     @CommandLine.Option(names = {"--package-version"}, description = "version of the Ballerina package")
     private String packageVersion;
 
+    @CommandLine.Option(names = {"--parent-package"}, description = "version of the Ballerina package")
```
Comment (sameeragunarathne) [inline native/health-cli/src/main/java/io/ballerina/health/cmd/fhir/FhirSubCmd.java:82]: Add the correct description

File: native/fhir-to-bal-lib/src/main/java/org/wso2/healthcare/fhir/ballerina/packagegen/tool/ToolConstants.java (hunk 1 above)
Comment (isuruh15) [reply native/fhir-to-bal-lib/src/main/java/org/wso2/healthcare/fhir/ballerina/packagegen/tool/ToolConstants.java:41]: No, base package refers to the FHIR base package which is mandatory for all packages.
The newly introduced parent package is for the cases where another package is referred as a dependancy and that is optional.

File: native/health-cli/src/main/java/io/ballerina/health/cmd/fhir/FhirSubCmd.java (hunk 3 above)
Comment (isuruh15) [reply native/health-cli/src/main/java/io/ballerina/health/cmd/fhir/FhirSubCmd.java]: fixed in 8e8bef2

File: native/fhir-to-bal-lib/src/main/java/org/wso2/healthcare/fhir/ballerina/packagegen/tool/modelgen/ResourceContextGenerator.java (hunk 2 above)
Comment (isuruh15) [reply native/fhir-to-bal-lib/src/main/java/org/wso2/healthcare/fhir/ballerina/packagegen/tool/modelgen/ResourceContextGenerator.java]: fixed in 8e8bef2

**Review by sameeragunarathne — APPROVED** [review]

**Pull request merged**